### DISCUSSION TITLE
Shelly Button

### DISCUSSION THREAD

**Discussion by scargill** [windkh/node-red-contrib-shelly#110] [Q&A] [answered]
Help a beginner to nod-red-contrib-shelly - I have a pair of Shelly buttons which I assume are version 1. Kind of works but I get messages about not being able to connect and timeouts unless the button is plugged in. So with a fully charged (but USB disconnected) button (fixed IP)...

![image](https://user-images.githubusercontent.com/1889411/217568749-6553fe6b-dc7d-4bf6-b728-615990a7bbaa.png)

I eterered the right IP address and with an RBE node to stop identical packages outputting - it works - showing "S" for single press etc and the battery state - but I get ping timeouts and ehost unreachable messages....  mode -polling - 1000ms interval - ticked to enable status output 

![image](https://user-images.githubusercontent.com/1889411/217569438-ae56ca2c-607d-477a-a03e-bbdb567fa583.png)

What am I doing wrong?

**Comment by windkh** [comment]
Buttons enter sleep mode and wake up only from time to time. In this case polling does not make sense. Change mode to callback and try again.

**Reply by scargill** [reply]
Callback version says "installing webhook"....
![image](https://user-images.githubusercontent.com/1889411/217570448-45fd707f-b43e-4cdc-9ff9-41dd64888d87.png)

![image](https://user-images.githubusercontent.com/1889411/217570635-ca58ae70-a220-4919-bfad-3a974fe5d824.png)

and doesnt product output...

**Reply by windkh** [reply]
When the button wakes up the webhook is installed. Just press the button to wake him up.

**Reply by scargill** [reply]
Crumbs that was quick. Went to callback - as you say - it connects when you predd the button but now no output on button press even if I connect debug directly to the node...

**Reply by windkh** [reply]
Hm ... does node red run inside a container?
Well, you can open the browser and enter the ip of your shelly. Check the Action settings. The node entered a webhook with an url pointing to your node-red server. 
1. is the ip of node red correct?
2. is something else blocking the callback? Firewall?

**Reply by scargill** [reply]
Node-Red isn't inside a container - it is running on a Raspberry Pi 4 - 192.168.1.19 - again fixed address. I opened my chrome browser and put in the address of the Shelly - initially nothing as you'd expect but then when I pressed the button - just blue screen and busy icon..

![image](https://user-images.githubusercontent.com/1889411/217575719-8a8db801-a7d7-4609-ae26-bd3bc1af2d9a.png)

Erm, I didn't put in the address of my RPI - am I being dumb here?

![image](https://user-images.githubusercontent.com/1889411/217576060-c511d580-9399-4075-9d2b-c93924408c3d.png)

**Reply by scargill** [reply]
Ok so I changed the 80 to 192.168.1.19:80 - made no difference, still no output.. Says connected on the node again.

**Reply by windkh** [reply]
The server configuration should contain a portnumber e.g. 10000 and the ip address of your node red server. 80 is not a good idea as this port is usually used by http servers.

**Reply by windkh** [reply]
Then check the IO-actions configuration via the shelly app please

**Reply by scargill** [reply]
So I changed the port to 10000 then looked at io actions in the shelly app...  

![image](https://user-images.githubusercontent.com/1889411/217591524-db3a7493-d776-49c3-aeb5-0f58a5e6c7d3.png)

and set as below

http://192.168.1.19:10000/webhook?data=shortpush_url?0?192.168.1.250

Still nothing out of the debug node when connected to the nrc-shelly node...
help - I'm really trying not to be thick here..

the 192.168.1.19 is definitely my RPI and 250 is my button.

**Reply by windkh** [reply]
Ok everythink seems to be fine.
Now copy the url in the webhook to your browser and trigger the callback from your computer just for crosschecking

http://192.168.1.19:10000/webhook?data=shortpush_url?0?192.168.1.250

Then please make sure that the debug node shows the complete msg not only payload so that we do not miss anything.

If calling the webhook from your browser window does not trigger the node, then you should probably inspect the node-red output on your rpi for errors.

**Reply by scargill** [reply]
Ah did you mean a browser on my RPI (I don;t have the graphics interface installed) - I am using a browser (chrome) on my PC....
![image](https://user-images.githubusercontent.com/1889411/217596385-2f18c82e-4cb9-44b1-8744-4f5e631dc8f4.png)

Just to be clear aftre I did the above (nothing in the browser but griping about "oage cannot be reached) and pressed the button on the shelly - I went back to the RPI NR and - nothing... 
![image](https://user-images.githubusercontent.com/1889411/217597289-fc03654c-ea9c-40cd-8ff2-bc7432b4e18c.png)

**Reply by windkh** [reply]
Ah the response states that the rpi refused to accept the call.

**Reply by scargill** [reply]
You suggested port 10000 - is there anything I need to do to the RPI to make it accept something to that port? My RPI has a web server running at port 86... 

![image](https://user-images.githubusercontent.com/1889411/217598716-68819fc3-d47b-4d60-964a-d318b42d97b8.png)

**Reply by scargill** [reply]
ALMOST. I removed the 19 address and just put in the port... NOW I get this when I press the button...

**Reply by windkh** [reply]
Ah ok I see. Glad that it works now for you. If you need further help or if you have an idea for a new useful feature, then let me know. Br kh

**Reply by scargill** [reply]
Will do. I'm going to get hold of a couple of V2 buttons to try those next.

**Comment by scargill** [comment, marked as answer]
all very pretty and no errors - but it doesn't tell me anything.
![image](https://user-images.githubusercontent.com/1889411/217600389-7e5e30a1-ca25-4c22-b18b-5bda24e969f8.png)

WRONG - my error -
I tried a double press...
![image](https://user-images.githubusercontent.com/1889411/217600607-605fd990-0514-48d6-89a5-db6425925bd9.png)


Yes, it DOES work - thank you SO much for your help. I'll leave you alone now....

**Reply by scargill** [reply]
Wheee.,,
![image](https://user-images.githubusercontent.com/1889411/217601672-4ab7b6ff-714c-42e7-baf1-8a031ab92aff.png)

**Reply by windkh** [reply]
Ok and what was finally the problem?

**Reply by scargill** [reply]
Me, basically.

**Comment by scargill** [comment]
One more stupid question - now I have that state coming out - working a treat - how can I add in the battery voltage? In my original method - the one with all the errors - I could get the battery voltage out?

**Reply by windkh** [reply]
In the configuration you can choose between event and status. Switch to status and you will get a different response.

**Reply by scargill** [reply]
Thank you so much. It seems that status is the better option as I can report the battery state and button state from the same object - lovely.
